### PR TITLE
CSHARP-4703: Initial implementation

### PR DESCRIPTION
**Pull request by sboulema**
Adds implementation for the scoreDetails feature

### CODE REVIEW

File: tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs [hunk 1]
```diff
@@ -280,6 +282,13 @@ public void Phrase()
 
             var highlightRangeStr = string.Join(string.Empty, highlightTexts.Skip(1).Select(x => x.Value));
             highlightRangeStr.Should().Be("Life, Liberty and the pursuit of Happiness.");
+
+            var scoreDetails = result.ScoreDetails.Value.Should().NotBe(0);
+
+            foreach (var scoreDetail in result.ScoreDetails.Details)
+            {
+                scoreDetail.Value.Should().NotBe(0);
+            }
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs:291]:
```suggestion

            result.ScoreDetails.Description.Should().Contain("life liberty and the pursuit of happiness");
            result.ScoreDetails.Value.Should().NotBe(0);

            var scoreDetail = result.ScoreDetails.Details.Should().ContainSingle().Subject;
            scoreDetail.Description.Should().NotBeNullOrEmpty();
            scoreDetail.Value.Should().NotBe(0);
            scoreDetail.Details.Should().NotBeEmpty();
```

**Review by BorisDog — CHANGES_REQUESTED** [review]
Looks good, thanks! Small suggestion for `Phrase` test.

File: src/MongoDB.Driver/Search/SearchScoreDetails.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+* limitations under the License.
+*/
+
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+
+namespace MongoDB.Driver.Search
+{
+    /// <summary>
+    /// A search count result set.
```
Comment (sanych-sun) [inline src/MongoDB.Driver/Search/SearchScoreDetails.cs:22]: This comment is seems to be wrong. Could you please update it?

**Review by sanych-sun — APPROVED** [review]
LGTM

Please fix the wrong comment

File: src/MongoDB.Driver/Search/SearchScoreDetails.cs (hunk 2 above)
Comment (sboulema) [reply src/MongoDB.Driver/Search/SearchScoreDetails.cs]: Good catch, copy & paste mistake! Comment updated.

**Review by BorisDog — APPROVED** [review]
LGTM!

**Pull request merged**